### PR TITLE
Init DB connections with correct modules

### LINKED ISSUE #130: `SyncState` endpoint: SQL error
On some calls to the `SyncState` RPC, we get the error

```
no such table: rarray Caused by: Error code 1: SQL error or missing database
```

It doesn't seem to happen on every call though.

### PR DESCRIPTION
**Pull request by igamigo**
Closes #130 
Only one connection from the pool was being initialized with the correct module, which caused incoming requests to randomly fail. This PR adds a hook that gets called on pool object creation that inits the connection correctly.

There might also be a better way to do this but skimming through the docs of the deadpool crate I did not find anything.

### CODE REVIEW

File: store/src/db/mod.rs [hunk 1]
```diff
@@ -49,25 +49,33 @@ impl Db {
...
+                        .interact(|conn| array::load_module(conn))
+                        .await
+                        .map_err(|_| HookError::StaticMessage("Loading carray module failed"))?;
+
+                    let _ = conn
+                        .interact(|conn| migrations::MIGRATIONS.to_latest(conn))
+                        .await
+                        .map_err(|_| {
+                            HookError::StaticMessage("Migration task failed with a panic")
+                        })?;
```
Comment (bobbinth) [inline store/src/db/mod.rs:67]: This probably doesn't need to happen for every connection - if so, we should keep it the way it was.

File: store/src/db/mod.rs [hunk 2]
```diff
@@ -49,25 +49,33 @@ impl Db {
...
-        let conn = pool.get().await?;
+        let pool = SqliteConfig::new(config.database_filepath.clone())
+            .builder(Runtime::Tokio1)?
+            .post_create(Hook::async_fn(move |conn, _| {
+                Box::pin(async move {
+                    // Feature used to support `IN` and `NOT IN` queries
+                    let _ = conn
+                        .interact(|conn| array::load_module(conn))
+                        .await
+                        .map_err(|_| HookError::StaticMessage("Loading carray module failed"))?;
```
Comment (bobbinth) [inline store/src/db/mod.rs:60]: I would add a comment here to explain that we need to do this on every connection when they are created.

**Review by bobbinth — CHANGES_REQUESTED** [review]
Looks good! Thank you! I left a couple of small comments inline. But also, could we verify that this indeed fixes the issue?

File: store/src/db/mod.rs (hunk 1 above)
Comment (igamigo) [reply store/src/db/mod.rs]: Agreed, changed it so it runs on one connection.

File: store/src/db/mod.rs [hunk 3]
```diff
@@ -49,24 +49,32 @@ impl Db {
...
-        // Feature used to support `IN` and `NOT IN` queries
-        conn.interact(|conn| array::load_module(conn))
-            .await
-            .map_err(|_| anyhow!("Loading carray module failed"))??;
+        let conn = pool.get().await?;
 
         conn.interact(|conn| migrations::MIGRATIONS.to_latest(conn))
             .await
-            .map_err(|_| anyhow!("Migration task failed with a panic"))??;
+            .map_err(|_| HookError::StaticMessage("Migration task failed with a panic"))??;
```
Comment (bobbinth) [inline store/src/db/mod.rs:77]: I would probably change this back to `anyhow` error - unless there is a good reason to use `HookError` here?

**Review by bobbinth — APPROVED** [review]
Looks good! Thank you! I left just one small comment inline.

**Review by plafer — APPROVED** [review]
LGTM! Works locally; tried a bunch of permutations of including/excluding `block_num` and `account_ids`

**Pull request merged**